rfcs/20190829-tfx-container-component-execution.md

```diff
@@ -0,0 +1,555 @@
+# TFX Container Component Execution Proposal
+
+Status         | Proposed
+:------------- | :-------
+**Author(s)**  | Ajay Gopinathan (<a href="mailto:ajaygopinathan@google.com">ajaygopinathan@google.com</a>), Hongye Sun (<a href="mailto:hongyes@google.com">hongyes@google.com</a>), Makoto Uchida (<a href="mailto:muchida@google.com">muchida@google.com</a>), Ruoyu Liu (<a href="mailto:ruoyu@google.com">ruoyu@google.com</a>)
+**Sponsor(s)** | Konstantinos Katsiapis (katsiapis@google.com), Pavel Dournov (dournov@google.com), Ramesh Chandra (rameshuc@google.com)
+**Updated**    | 2019-08-29
+
+## Objective
+
+This RFC proposes an orchestrator agnostic way to reliably execute a user’s
+container in the TFX pipeline. The proposal can support:
+
+*   Running an arbitrary container in either local docker environment or remote
+    k8s cluster.
+*   Passing data into the container
+*   Passing output data from the container
+*   Capturing logs from the container
+*   Handling errors and retries
+*   Cancelling the container execution if the pipeline is terminated
```

Comment:
Could you elaborate more on what "Cancelling" means including a timeline of what happens later in this document?  For instance, are their provisions to signal the running container so that it can perform a clean shutdown?

Reply:
Container Lifetime:  Container instances may benefit from internal caching or reductions in startup time (ie applications within container taking some time to start up).  Ideally, the container instance could stay alive across multiple pipeline calls of it.